dev-tools/omdb/src/bin/omdb/db/saga.rs

```diff
@@ -0,0 +1,182 @@
+    command: SagaCommands,
+}
+
+#[derive(Debug, Subcommand, Clone)]
+enum SagaCommands {
+    /// List running sagas
+    Running,
+
+    /// Inject an error into a saga's currently running node
+    Fault(SagaFaultArgs),
+}
+
+#[derive(Clone, Debug, Args)]
+struct SagaFaultArgs {
+    saga_id: Uuid,
+}
+
+impl SagaArgs {
+    pub async fn exec(
+        &self,
+        omdb: &Omdb,
+        opctx: &OpContext,
+        datastore: &DataStore,
+    ) -> Result<(), anyhow::Error> {
+        match &self.command {
+            SagaCommands::Running => cmd_sagas_running(opctx, datastore).await,
+
+            SagaCommands::Fault(args) => {
+                let token = omdb.check_allow_destructive()?;
+                cmd_sagas_fault(opctx, datastore, args, token).await
+            }
+        }
+    }
+}
+
+async fn cmd_sagas_running(
+    _opctx: &OpContext,
+    datastore: &DataStore,
+) -> Result<(), anyhow::Error> {
+    let conn = datastore.pool_connection_for_tests().await?;
+
+    let sagas =
+        get_all_sagas_in_state(&conn, steno::SagaCachedState::Running).await?;
+
+    #[derive(Tabled)]
+    struct SagaRow {
+        id: Uuid,
+        creator_id: Uuid,
+        time_created: String,
+        name: String,
+        state: String,
+    }
+
+    let rows: Vec<_> = sagas
+        .into_iter()
+        .map(|saga: Saga| SagaRow {
+            id: saga.id.0.into(),
+            creator_id: saga.creator.0,
+            time_created: datetime_rfc3339_concise(&saga.time_created),
+            name: saga.name,
+            state: format!("{:?}", saga.saga_state),
+        })
+        .collect();
+
+    let table = tabled::Table::new(rows)
+        .with(tabled::settings::Style::psql())
+        .to_string();
+
+    println!("{}", table);
+
+    Ok(())
+}
+
+async fn cmd_sagas_fault(
+    _opctx: &OpContext,
+    datastore: &DataStore,
+    args: &SagaFaultArgs,
+    _destruction_token: DestructiveOperationToken,
+) -> Result<(), anyhow::Error> {
+    let conn = datastore.pool_connection_for_tests().await?;
+
+    // Find the most recent node for a given saga
+    let most_recent_node: SagaNodeEvent = {
+        use db::schema::saga_node_event::dsl;
+
+        dsl::saga_node_event
+            .filter(dsl::saga_id.eq(args.saga_id))
+            .order(dsl::event_time.desc())
+            .limit(1)
+            .first_async(&*conn)
+            .await?
+    };
+
+    // Inject a fault for that node, which will cause the saga to unwind
+    let action_error = steno::ActionError::action_failed(String::from(
+        "error injected with omdb",
+    ));
+
+    let fault = SagaNodeEvent {
+        saga_id: most_recent_node.saga_id,
+        node_id: most_recent_node.node_id,
+        event_type: steno::SagaNodeEventType::Failed(action_error.clone())
+            .label()
+            .to_string(),
+        data: Some(serde_json::to_value(action_error)?),
+        event_time: chrono::Utc::now(),
+        creator: most_recent_node.creator,
+    };
+
```

Comment:
Can we print out exactly what it's doing?  Something like:

```
injecting fault into running node ${node_id}
```

Reply:
done in b1108a38a